tool/src/config.rs

```diff
@@ -262,3 +263,15 @@ pub(crate) fn find_top_level_attr(module_items: Vec<syn::Item>) -> Vec<DiplomatB
 
     out_config
 }
+
+#[cfg(test)]
+mod test {
+    use toml::Value;
+
+    #[test]
+    fn test_toml_parse() {
+        let t = "true";
+        assert!(toml::from_str::<Value>(t).is_err());
+        assert_eq!(super::toml_value_from_str(t), Value::String(t.to_string()));
```

Comment:
I tried to test my new scaffold argument and saw that it didn't actually work, which led down to a rabbit hole.

Here we see that the value `true` fails to parse as toml, and so in `toml_value_from_str` we fall back to a string.

I've a got a test which, I think, simplifies all of the config. Instead of manually parsing all of arguments, we can parse arguments like `kotlin.scaffold=true` and it (conceptually) will result in `Table { "kotlin": Table { "scaffold": true}}`. Then we can extend the table parsed from the config filie, and just parse the full `Config` type from this untyped toml `Table` (requires `toml_edit` as a dependency.

I'll create an issue explaining in more detail.